OctoPrintLib/Main Classes/OctoprintConnection.cs

```diff
@@ -65,25 +65,43 @@ protected async Task<string> PostStringAsync(string location, string arguments)
         /// <param name="arguments">The Newtonsoft Jobject to post tp the address</param>
         protected string PostJson(string location, JObject arguments)
         {
-            string strResponseValue = string.Empty;
-            String argumentString = string.Empty;
-            argumentString = JsonConvert.SerializeObject(arguments);
-            HttpWebRequest request = WebRequest.CreateHttp("http://" + server.DomainNmaeOrIp +"/"+ location);// + "?apikey=" + apiKey);
-            request.Method = "POST";
-            request.Headers["X-Api-Key"] = server.ApplicationKey;
-            request.ContentType = "application/json";
-            using (var streamWriter = new StreamWriter(request.GetRequestStream()))
+            try
             {
-                streamWriter.Write(argumentString);
-            }
-            HttpWebResponse httpResponse;
-            httpResponse = (HttpWebResponse)request.GetResponse();
+                string strResponseValue = string.Empty;
+                String argumentString = string.Empty;
+                argumentString = JsonConvert.SerializeObject(arguments);
+                HttpWebRequest
+                    request = WebRequest.CreateHttp("http://" + server.DomainNmaeOrIp + "/" +
+                                                    location); // + "?apikey=" + apiKey);
+                request.Method = "POST";
+                request.Headers["X-Api-Key"] = server.ApplicationKey;
+                request.ContentType = "application/json";
+                using (var streamWriter = new StreamWriter(request.GetRequestStream()))
+                {
+                    streamWriter.Write(argumentString);
+                }
 
-            using (var streamReader = new StreamReader(httpResponse.GetResponseStream()))
+                HttpWebResponse httpResponse;
+                httpResponse = (HttpWebResponse) request.GetResponse();
+
+                using (var streamReader = new StreamReader(httpResponse.GetResponseStream()))
+                {
+                    strResponseValue = streamReader.ReadToEnd();
+                }
+
+                return strResponseValue;
+            }
+            catch (WebException e)
             {
-                strResponseValue = streamReader.ReadToEnd();
+                string msg =
+                    "The Octoprint environment was not set up correctly. Consider checking that Octoprint is already running " +
```

Comment:
Start with a short description of the error like: The connection to Octoprint could not be established.